functions/consumer/wavefront-consumer/src/main/java/org/springframework/cloud/fn/consumer/wavefront/WavefrontConsumerProperties.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.consumer.wavefront;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.Pattern;
+import javax.validation.constraints.Size;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.validation.annotation.Validated;
+
+/**
+ * @author Timo Salm
+ */
+@ConfigurationProperties("wavefront")
+@Validated
+public class WavefrontConsumerProperties {
+
+	private String metricName;
+
+	private String source;
+
+	private String metricJsonPath;
```

Comment:
We should user SpEL expression (and it #jsonPath support) rather than explicitly call JsonPath utilities. 
Furthermore the SpEL expression properties are more generic and can be used against any payload type not only JSON.